### PR TITLE
Add option to use entry only for HTTP Basic Auth

### LINKED ISSUE #3925: Add HttpAuth flag
Please add an `HttpAuth` flag as discussed at https://github.com/keepassxreboot/keepassxc-browser/pull/628.

### PR DESCRIPTION
**Pull request by varjolintu**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
Adds a new option to use entry only with HTTP Basic Auth. If an entry uses additional URL's, the option will be enabled for them also.

Fixes #3925.

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
Screen Shot 2019-11-24 at 18 01 02

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Manually.

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/gui/entry/EditEntryWidgetBrowser.ui [hunk 1]
```diff
@@ -50,6 +50,13 @@
         </property>
        </widget>
       </item>
+      <item>
+       <widget class="QCheckBox" name="onlyHttpAuthCheckbox">
+        <property name="text">
+         <string>Use this entry only with HTTP Basic Auth</string>
```
Comment (droidmonkey) [inline src/gui/entry/EditEntryWidgetBrowser.ui]: We should add a tooltip to this that expands on the setting a little more. Such as:

> Only send this setting to the browser for HTTP Auth dialogs. If enabled, normal login forms will not show this entry for selection.

**Comment by varjolintu** [comment]
> If there are two entries for a webpage, one of them has this setting, is the following true:
> 
>     1. Login form only shows the entry that does NOT have this setting
> 
>     2. HTTP Auth form shows BOTH entries

2. Both entries are showed.

**Comment by droidmonkey** [comment]
Is that the desired behavior? Or should be only show the ones marked explicitly for http auth (if any are marked as such).

File: src/gui/entry/EditEntryWidget.cpp [hunk 2]
```diff
@@ -971,6 +975,12 @@ void EditEntryWidget::setForms(Entry* entry, bool restore)
         m_browserUi->hideEntryCheckbox->setChecked(false);
     }
 
+    if (m_customData->contains(BrowserService::OPTION_ONLY_HTTP_AUTH)) {
+        m_browserUi->onlyHttpAuthCheckbox->setChecked(m_customData->value(BrowserService::OPTION_ONLY_HTTP_AUTH) == "true");
```
[the remark this answers is not in the document]
Comment (varjolintu) [reply src/gui/entry/EditEntryWidget.cpp]: Using this in Global.h would collide with default defines, e.g. `#define FALSE 0`. Had to rename the variables.

[the remark this answers is not in the document]
Comment (droidmonkey) [reply src/gui/entry/EditEntryWidget.cpp]: Perhaps TRUE_STR/FALSE_STR?

**Comment by varjolintu** [comment]
@droidmonkey Yes. The whole idea was to hide entries from other than HTTP Basic Auth. This is also easy to test if you add the option to an entry and add an Additional URL to some other site.

File: src/gui/entry/EditEntryWidget.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (varjolintu) [reply src/gui/entry/EditEntryWidget.cpp]: Maybe the first would be a little bit better. For some reason the TC build log is now filled with `Global.h:25:10: fatal error: 'QStringLiteral' file not found` even if everything works fine on my local machine.

[the remark this answers is not in the document]
Comment (droidmonkey) [reply src/gui/entry/EditEntryWidget.cpp]: You might need to explicitly define the header file for it. This is a problem with older Qt versions which the CI runs on.

[the remark this answers is not in the document]
Comment (varjolintu) [reply src/gui/entry/EditEntryWidget.cpp]: That is already defined. Or have I missed something?

**Pull request merged**